sql/core/src/test/java/test/org/apache/spark/sql/execution/sort/RecordBinaryComparatorSuite.java

```diff
@@ -321,4 +321,69 @@ public void testBinaryComparatorWhenOnlyTheLastColumnDiffers() throws Exception
 
     assert(compare(0, 1) < 0);
   }
+
+  @Test
+  public void testBinaryComparatorGiveSameResultWhenComparedByteByByteAndComparedByLong()
```

Comment:
Heh, you might chop this down a bit. `testCompareLongsAsLittleEndian()` or something.